src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.commons.io
+
+import java.nio.file.{Files, Path}
+
+import com.fulcrumgenomics.commons.util.LazyLogging
+
+import scala.collection.mutable
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+trait CommandLineTool extends LazyLogging {
+  /** The name of the executable such as Rscript or gs. */
+  val Executable: String
+
+  /** Command used for the above executable. */
+  val TestCommand: Seq[String]
+
+  /** Exception class that holds onto an executable's exit/status code. */
+  case class ToolException(status: Int) extends RuntimeException {
+    override def getMessage: String = s"$Executable failed with exit code $status."
+  }
+
+  /** Returns a tuple of (Boolean, String): first element is true when command can execute
+    * with no error; second element is a string of command stdout */
+  def execArgs(command: String*): (Boolean, String) = {
+    try {
+      // Read redirected stdout to a string
+      val process = new ProcessBuilder(command: _*).redirectErrorStream(true).start()
+      val stdout  = mutable.ListBuffer[String]()
+      val sink    = new AsyncStreamSink(process.getInputStream, s => stdout.append(s))
+      // True if the command can execute with no error
+      val canExecute = process.waitFor() == 0
+      sink.close()
+
+      (canExecute, stdout.toList.mkString)
+    }
+    catch {case e: Exception => (false, "")}
+  }
+
+  /** Returns true if the tool is available and false otherwise. */
+  lazy val Available: Boolean = {
+    execArgs(Executable +: TestCommand: _*)._1
```

Comment:
If `execArgs` returns a `Try[String]` then this lazy trait constant becomes:

```suggestion
    execArgs(Executable +: TestArgs: _*).isSuccess
```

Which is more readable.